### PR TITLE
Gated Pg/ Gartner report

### LINKED ISSUE #5369: Gated page: Gartner report
Sourcegraph has officially been recognized by Gartner as a “Cool Vendor” in the “Software Engineering: Enhancing Developer Productivity” category. We are creating a page where users can download the report for free.

⚠️IMPORTANT: Please don't publish until you have confirmation from Andy or Fabiana ⚠️ Tentative publish date: June 1 
 
[🔗Meta](https://docs.google.com/spreadsheets/d/1qF9ztZ2my-sX8B5pdlZOFvAEb9TzB0DgiKM9yVkIM14/edit#gid=1572625403)
🔗[Figma](https://www.figma.com/file/vVvebR510dGnR8zPx12Upv/Gartner?node-id=0%3A1)

After the user fills in the form, they will be directed to https://www.gartner.com/reprints/?id=1-2A44UTV8&ct=220524&st=sb

```
<!--[if lte IE 8]>
<script charset="utf-8" type="text/javascript" src="//js.hsforms.net/forms/v2-legacy.js"></script>
<![endif]-->
<script charset="utf-8" type="text/javascript" src="//js.hsforms.net/forms/v2.js"></script>
<script>
 hbspt.forms.create({
 region: "na1",
 portalId: "2762526",
 formId: "4f5b127f-4d42-490a-a970-686574d0e542"
});
</script>
```

Asana: https://app.asana.com/0/1200095757619647/1202286036290353/f

### PR DESCRIPTION
**Pull request by st0nebraker**
This closes #5369. It adds a new gated report page using our new simple gated page layout

### Changelog
- Adds `/2022-gartner-cool-vendors-in-software-engineering`

### Note
- This merges into the open Guide/vulernabilities PR to illustrate the limited updates. Once that PR is merged, GH automatically changes the merge destination to: `main`

### Test
1. Navigate to `/2022-gartner-cool-vendors-in-software-engineering`
2. Please check that layout matches Figma specs
3. Ensure linting passes.
4. Ensure prettier has standardized the proposed changes.
5. Ensure dev and production builds work.

### CODE REVIEW

File: src/pages/2022-gartner-cool-vendors-in-software-engineering.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            formLabel="Download the report"
+            form={{
+                formId: '4f5b127f-4d42-490a-a970-686574d0e542',
+                onFormSubmitted: () =>
+                    window.open('https://www.gartner.com/doc/reprints?id=1-2A44UTV8&ct=220524&st=sb'),
+            }}
+            description={
+                <section className="col-md-6 col-12 pr-lg-6">
+                    <h3 className="font-weight-normal pb-5">
+                        Get complimentary access to 2022 Gartner® Cool Vendors™ report to learn how we believe
```
Comment (zlonko) [inline src/pages/2022-gartner-cool-vendors-in-software-engineering.tsx:29]: Side note: The screen reader announced the special characters correctly 👌 I was curious if they would without the encoding (like `&#8482;`). TIL that the UTF-8 character is preferred for readability, like you did here! Issues can happen when the font has insufficient coverage, but this isn't the case here.

**Review by zlonko — APPROVED** [review]
This looks good: I am able to submit the form and open the page as expected. I left a note in the Figma anticipating a comment about the copy from our stakeholders.

File: src/components/SimpleGatedPg/index.tsx [hunk 2]
```diff
@@ -95,22 +94,44 @@ export const SimpleGatedPgLayout: FunctionComponent<Props> = ({
...
+                    <div
+                        className={classNames(
+                            'col-12',
+                            customer && isWebinarPg && 'col-lg-8',
+                            customer && !isWebinarPg && 'col-lg-10'
+                        )}
+                    >
                         <h1 className="display-2 font-weight-bold mb-4 mb-md-2">{title}</h1>
                         {subtitle && <h3 className="font-weight-normal max-w-800">{subtitle}</h3>}
                     </div>
```
[the remark this answers is not in the document]
Comment (st0nebraker) [reply src/components/SimpleGatedPg/index.tsx:137]: Yes, forgot about this piece. Thanks for handling that for me Tim! 🙌

File: src/pages/2022-gartner-cool-vendors-in-software-engineering.tsx (hunk 1 above)
Comment (st0nebraker) [reply src/pages/2022-gartner-cool-vendors-in-software-engineering.tsx:29]: Woohoo!

File: src/pages/2022-gartner-cool-vendors-in-software-engineering.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                        not endorse any vendor, product or service depicted in its research publications and does not
+                        advise technology users to select only those vendors with the highest ratings or other
+                        designation. Gartner research publications consist of the opinions of Gartner’s Research &
+                        Advisory organization and should not be construed as statements of fact. Gartner disclaims all
+                        warranties, expressed or implied, with respect to this research, including any warranties of
+                        merchantability or fitness for a particular purpose.
+                    </p>
+                    <p>
+                        Gartner, Cool Vendors in Software Engineering: Enhancing Developer Productivity, Cool Vendors in
+                        Software Engineering: Enhancing Developer Productivity, 16th May 2022.
```
[the remark this answers is not in the document]
Comment (zlonko) [reply src/pages/2022-gartner-cool-vendors-in-software-engineering.tsx:44]: Update: Fabiana responded, confirming that this is the correct copy.

**Review by bretthayes — DISMISSED** [review]

**Comment by st0nebraker** [comment]
This is ready for stakeholder review @elzannewentzel !

**Review by bretthayes — APPROVED** [review]

**Pull request merged**